### PR TITLE
Fixes compatibility with NodeJS 18 fetch implementation

### PR DESCRIPTION
**Pull request by Tpt**
- The response body is a WebStream and not a NodeJS Stream
- The "url" option is not allowed in Response constructor: https://developer.mozilla.org/en-US/docs/Web/API/Response/Response#options

### CODE REVIEW

File: lib/Util.ts [hunk 1]
```diff
@@ -161,7 +167,7 @@ export class Util {
           writeStream.on('close', resolve);
           writeStream.on('error', reject);
         });*/
-        writeFileSync(cachePathLocal + '.url', response.url);
```
Comment (rubensworks) [inline lib/Util.ts:164]: We need the final URL though, since this URL might have changed with redirects.

File: lib/Util.ts [hunk 2]
```diff
@@ -170,7 +176,7 @@ export class Util {
       return {
         body: body2,
         headers: response.headers,
-        url: response.url,
```
Comment (rubensworks) [inline lib/Util.ts:173]: Here as well.

**Review by rubensworks — CHANGES_REQUESTED** [review]
Perhaps we can override the URL of responses in tests in a different way?

File: lib/Util.ts (hunk 1 above)
Comment (Tpt) [reply lib/Util.ts:164]: Fixed.

**Pull request merged**